src/plugins/data/public/antlr/opensearch_ppl/code_completion.ts

```diff
@@ -35,16 +37,48 @@ export const getSuggestions = async ({
     const finalSuggestions: QuerySuggestion[] = [];
 
     if (suggestions.suggestColumns) {
-      finalSuggestions.push(...fetchFieldSuggestions(indexPattern));
+      finalSuggestions.push(...fetchFieldSuggestions(indexPattern, (f: any) => `${f} `));
     }
 
+    if (suggestions.suggestAggregateFunctions) {
+      finalSuggestions.push(
+        ...PPL_AGGREGATE_FUNTIONS.map((af) => ({
+          text: `${af}()`,
+          type: monaco.languages.CompletionItemKind.Function,
+          insertText: af + '(${1:expr}) ',
+          detail: SuggestionItemDetailsTags.AggregateFunction,
+          insertTextRules: monaco.languages.CompletionItemInsertTextRule.InsertAsSnippet,
+        }))
+      );
+    }
+
+    if (suggestions.suggestSourcesOrTables) {
+      finalSuggestions.push({
+        text: indexPattern.title,
+        type: monaco.languages.CompletionItemKind.Struct,
+        insertText: `${indexPattern.title} `,
+        detail: SuggestionItemDetailsTags.Table,
+      });
+    }
+
+    // create the sortlist
+    const suggestionImportance = new Map<number, string>();
+    suggestionImportance.set(OpenSearchPPLParser.PIPE, '0');
+    suggestionImportance.set(OpenSearchPPLParser.COMMA, '1');
+    suggestionImportance.set(OpenSearchPPLParser.PLUS, '2');
+    suggestionImportance.set(OpenSearchPPLParser.MINUS, '2');
+    suggestionImportance.set(OpenSearchPPLParser.EQUAL, '2');
+    suggestionImportance.set(OpenSearchPPLParser.SOURCE, '2');
+
     // Fill in PPL keywords
     if (suggestions.suggestKeywords?.length) {
       finalSuggestions.push(
         ...suggestions.suggestKeywords.map((sk) => ({
-          text: sk.value,
+          text: sk.value.toLowerCase(),
+          insertText: `${sk.value.toLowerCase()} `,
```

Comment:
why lowercase? for PPL keywords do we want them to be in upper case to distinguish them from variables?

Reply:
I was just following the convention that existed in log explorer. Do we want to change it? The current placeholder when selecting a dataset is `source = <index>` so I was just matching that as well, we would need to change that.